website/blog/2022-10-31-simple-1.mdx

```diff
@@ -0,0 +1,102 @@
+---
+slug: simple-functions-1
+title: "Simple Functions Introduction and Basic Optimizations"
+authors: [lsakka]
+tags: [tech-blog, simple-functions]
+---
+
+_This blogpost is part of a series of blog posts that discuss different features and optimizations of the simple function interface_.
+
+
+## Introduction to simple functions
+
+Scalar functions are one of the most used extension points in Velox. Since Velox is a vectorized engine, by nature functions are <a href="https://facebookincubator.github.io/velox/develop/scalar-functions.html#vector-functions">"vector functions"</a> that consume vectors and produce vectors. Velox allows users to write functions as vector functions or as single-row operations "simple functions" that are converted to vector functions using template expansion through <a href="https://github.com/facebookincubator/velox/blob/main/velox/expression/SimpleFunctionAdapter.h">SimpleFunctionAdapter</a>.
+
+<figure>
+<img src="/img/simple1_1.png"/>
+</figure>
+
+The user only needs to implement the PlusFunction struct shown in the graph above, which is then expanded using template expansion to a vector function.
```

Comment:
Maybe we could expand a bit on the motivation of simple functions? (vast majority can be expressed using this interface, much easier and more natural to developers, less error prone). Maybe we could even say that the motivation came from developers complaining the API was too complex.

Reply:
I think we need to agree on pitch here, @mbasmanova  @pedroerp